src/main/webapp/resources/js/apis/metadata/field.ts

```diff
@@ -49,6 +61,27 @@ export const {
   useUpdateProjectMetadataFieldRestrictionMutation,
 } = fieldsApi;
 
+/**
+ * Get a list of fields for the project
+ * @returns {Promise<any>}
+ */
+export async function getMetadataFieldsForProject(projectId: string) {
+  try {
```

Comment:
I just wanted to take a minute and explain this a little more.  There is a rule available in eslint (`explicit-function-return-type`) to check that **every** function has an explicit return type stated.  This is not what we want because it really is overkill.  Typescript has the ability to infer from the item return from the function what the return type should be, and in your editor, you should be able to hover over the function name and see this (I will show you more below.

On line 66, there is a JSDoc typing for `Promis<any>`, we do not want to put any types into the JSDoc, just describe what the parameter is for.  If we hover over the function name in the IDE we see the following:

![](https://user-images.githubusercontent.com/11295750/214603328-588ef78d-97c7-4d01-8f32-7e181719a64e.png)

You can see that currently the IDE, using typescript, is inferring what the return type should be `Promise<AxiosResponse<any, any>>`.

We can figure out the proper return type by looking at the axios typing definition for `POST`:

```typescript
post<T = any, R = AxiosResponse<T>, D = any>(url: string, data?: D, config?: AxiosRequestConfig<D>): Promise<R>;
```

Here we see that you can type the request, and honestly, from looking at how other people (e.g. [https://bobbyhadz.com/blog/typescript-http-request-axios)](https://bobbyhadz.com/blog/typescript-http-request-axios)) do it, this is what I came up with this (please not that I am arbitrarily using a `number` here since this actually does not return anything, this is for demonstration purposes only):

![](https://user-images.githubusercontent.com/11295750/214611116-cb5cfa32-fac5-4405-addd-91e679a98160.png)

Typescript is now inferring that you are returning `Promise<AxiosPromise<number, any>>`, the final `any`, in this case, is actually ok, as I will show next.

If we look at the caller (again modified to return a number), we see that if we destructure the `AxiosResponse` and just get `data`, it is properly typed now as a `number`:

![](https://user-images.githubusercontent.com/11295750/214611886-3c1851ed-894c-478f-92f9-c641ba437aea.png)

Therefore, explicitly forcing every function in eslint to have a return type is not necessary, but does take a little work by all of us to my sure the type we return from a function is what we are expecting.

@ericenns  and I have started working on standardizing these calls and making it a little easier in the file `requests.ts`, although there is currently an issue that @deepsidhu85 is fixing for me.  The new format will be:

```typescript
post<returnType, parameterType>(url)
```

This will handle unwrapping the axios response and errors.

Reply:
Is this closer to what you're thinking? Please see 5c50179c42c3b63606dcaff51dd5beac557eb178.

Reply:
That is exactly it.  Now you will see if you hover over the function name you see the proper return type.  Since the function is typed properly, in your `importReducer` on line 243, you can actually remove typing the response because it is now known. (You don't actually need to do this I am just demonstrating the power of properly typing the return value).  It really simplifies the code.  Thanks for updating this.

![](https://user-images.githubusercontent.com/11295750/215125895-cf296c28-add7-4cd2-a568-4588ebf05e4c.png)